lib/src/mock_query.dart

```diff
@@ -71,52 +71,100 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
     parameters['orderedBy'].add(field);
     return MockQuery(this, (docs) {
       final sortedList = List.of(docs);
-      sortedList.sort((d1, d2) {
-        dynamic value1;
-        if (field is String) {
-          value1 = d1.get(field) as Comparable;
-        } else if (field == FieldPath.documentId) {
-          value1 = d1.id;
-        }
-        dynamic value2;
-        if (field is String) {
-          value2 = d2.get(field);
-        } else if (field == FieldPath.documentId) {
-          value2 = d2.id;
-        }
-        if (value1 == null && value2 == null) {
-          return 0;
-        }
-        // Return null values first.
-        if (value1 == null) {
-          return -1;
-        }
-        if (value2 == null) {
-          return 1;
-        }
-        final compare = value1.compareTo(value2);
-        return descending ? -compare : compare;
-      });
+      final fields = (parameters['orderedBy'] ?? []);
+      for (var index = 0; index < fields.length; index++) {
```

Comment:
Is this necessary to fix 217? Can you explain what needed rework?

Reply:
This PR is from 3 months ago and the code is not exactly straight forward at every turn. This might be from one of the PR this one was chained onto..

Reply:
Thank you! And sorry for taking so long to review. It makes us forget the context. That's why I merged #216 as is, since it fixed the issue, then cleaned up the code myself.

Reply:
For info I'm really busy right now. This is still on my mind, but I have priorities, I don't think I will be able to come back to this in the immediate future.